providers/config_default.yaml

```diff
@@ -155,6 +155,18 @@ VSPHERE_AZ_0:
 VSPHERE_AZ_1:
 VSPHERE_AZ_2:
 
+#! Settings to provide netpermissions for vpshere-csi file volumes
+#! example:
+#! NETPERMISSIONS:
+#!   A:
+#!     ips: "*"
+#!     permissions: READ_WRITE
+#!     rootsquash: false
+NETPERMISSIONS:
```

Comment:
A map object does not work here and you can expose 3 variables.

Reply:
will update as per suggestion

Reply:
@DanielXiao @nikhilbarge this is a typical case that the addon config will include a map struct. I do not think it should be exposed to config file with 3 variables. Instread, workload cluster users should create a new clusterbootstrap to include this optional netpermissions when creating a workload cluster, or edit the vpsherecsiconfig to include this optional netpermissions after creating a management cluster or a workload cluster. + @shyaamsn @adduarte @12345lcr

Reply:
Got it, it is a map[string]*NetPermissionConfigs type. We can also use a legacy style variable NETPERMISSIONS_BASE64 to encode the yaml object then decode it to the field netPermissions. It is also acceptable if it is not supported through legacy configuration file.